qiskit_nature/second_q/operators/sparse_label_op.py

```diff
@@ -32,6 +32,16 @@
 from .polynomial_tensor import PolynomialTensor
 
 
+_TCoeff = Union[complex, ParameterExpression]
+
+
+def _to_number(a: SupportsComplex | ParameterExpression) -> complex | float:
```

Comment:
```suggestion
def _to_number(a: SupportsComplex | ParameterExpression) -> complex:
```
`complex` implies `float` so this type hint union should not be necessary